### PR TITLE
[GEOT-6370] App-Schema Cardinality 1..N support for unbounded anonymous sequence

### PR DESCRIPTION
**Pull request by fernandor777**
This PR adds Cardinality 1..N support for unbounded anonymous sequence children elements, using ClientProperty tag.

https://osgeo-org.atlassian.net/browse/GEOT-6370

### CODE REVIEW

File: modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java [hunk 1]
```diff
@@ -59,185 +72,192 @@ public void setContext(MutablePicoContainer context) {
...
-                    if (values.size() > 1) {
-                        // make a multi property
-                        particle.setMaxOccurs(-1);
+                    // wrap the property in a particle
+                    XSDParticle particle = XSDFactory.eINSTANCE.createXSDParticle();
+                    XSDElementDeclaration wrapper =
+                            XSDFactory.eINSTANCE.createXSDElementDeclaration();
+                    wrapper.setResolvedElementDeclaration(elementDecl);
+                    particle.setContent(wrapper);
+                    // particle.setContent(elementDecl);
```
Comment (nmco) [inline modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java]: Letfover?

**Comment by nmco** [comment]
Generally speaking looks good, clean code and test cases implemented in [GeoServer](https://github.com/geoserver/geoserver/pull/3776).

**Comment by nmco** [comment]
That say, I have a few comments about how the new options are expressed in App-Schema configuration, I see several issues, the most relevant are:
- ``ClientProperty`` properties are for XML attributes, not elements, this makes the configuration quite confusing, and anonymous elements may have attributes too (if I'm not mistaken)
- the anonymous element may have another nested anonymous element

Consider the following example:

```xml
<xs:complexType name="StationType">
  <xs:complexContent>
    <xs:extension base="gml:AbstractFeatureType">
      <xs:sequence>
        <xs:element name="name" type="xs:string"/>
        <xs:element name="contact" type="st:ContactPropertyType"/>
        <xs:element name="location" type="gml:GeometryPropertyType"/>
        <xs:element maxOccurs="unbounded" minOccurs="0" name="tag" type="st:TagType"/>
        <xs:element name="measurements" type="ms:MeasurementPropertyType" minOccurs="0" maxOccurs="unbounded"/>
        <xs:element name="maintainer" minOccurs="0">
          <xs:complexType>
            <xs:sequence maxOccurs="unbounded">
              <xs:element name="name" type="xs:string"/>
              <xs:element name="level" type="xs:integer" nillable="true"/>
              <xs:element name="classType">
              <xs:complexType>
                  <xs:sequence maxOccurs="unbounded">
                    <xs:element name="name" type="xs:string"/>
                    <xs:element name="code" type="xs:integer"/>
                  </xs:sequence>
                </xs:complexType>
              </xs:element>
            </xs:sequence>
          </xs:complexType>
        </xs:element>
      </xs:sequence>
    </xs:extension>
  </xs:complexContent>
</xs:complexType>
```
I would suggest the following configuration:

```xml
(...)
<AttributeMapping>
  <targetAttribute>st:maintainer</targetAttribute>
  <anonymousAttribute>
    <attribute>st:name</name>
    <value>LEVEL</value>
  </anonymousAttribute>
  <anonymousAttribute>
    <attribute>st:level</name>
    <value>LEVEL</value>
  </anonymousAttribute>
  <anonymousAttribute>
    <attribute>st:classType/st:name</name>
    <value>CLASS_TYPE_NAME</value>
  </anonymousAttribute>
    <anonymousAttribute>
    <attribute>st:classType/st:code</name>
    <value>CLASS_TYPE_CODE</value>
  </anonymousAttribute>
  <jdbcMultipleValue>
    <sourceColumn>ID</sourceColumn>
    <targetTable>MAINTAINERS</targetTable>
    <targetColumn>OWNER</targetColumn>
  </jdbcMultipleValue>
</AttributeMapping>
(...)
```

What do you think @fernandor777?

File: modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java (hunk 1 above)
Comment (fernandor777) [reply modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/impl/BindingPropertyExtractor.java]: Looks like original code leftover, of course we can clean it.

**Comment by fernandor777** [comment]
Thanks Nuno for your review, indeed using a new "anonymousAttribute" tag definitions is a better idea and far more future proof, so I changed the PR to use them.

**Comment by nmco** [comment]
Great, thx @fernandor777, I didn't had a data set to manually test this, but the GeoServer integration tests look quite complete, so I'm assuming is working well 😄.

**Pull request merged**